### ISSUE TITLE
spmv, spmv_mv, spmv_struct, spgemm_jacobi fail on CUDA+ float, complex_float

### ISSUE THREAD

**Issue by brian-kelley** [kokkos/kokkos-kernels#715]
@seheracer @lucbv 

The "all scalars" build on Jenkins is finishing now, but there are some sparse test failures for float and complex_float. Just seems to be precision related, so maybe just a higher tolerance is the fix. I think @lucbv already knows about the spmv ones, but the spgemm_jacobi is new.

**Comment by brian-kelley** [comment]
The build on Jenkins: https://jenkins-son.sandia.gov/view/KokkosKernels/job/KokkosKernels_KokkosDev2_CUDA10_1_AllScalars/

**Comment by seheracer** [comment]
Should I work on this problem on kokkos-dev-2? @srajama1 

I don't think I have an account for the link @brian-kelley provided.

**Comment by brian-kelley** [comment]
@seheracer Yes, this is from kokkos-dev2, but they might fail on other machines.
Here's the build command:
```
./kokkos-kernels/scripts/cm_test_all_sandia cuda/10.1 --build-list=Cuda_OpenMP \
--with-scalars=double,float,complex_double,complex_float
```

**Comment by srajama1** [comment]
Every Sandian should be able to access jenkins-son. May be you need to VPN in ?

Like @brian-kelley said you might be able to reproduce on other machines.

**Comment by seheracer** [comment]
> Every Sandian should be able to access jenkins-son. May be you need to VPN in ?
> 
> Like @brian-kelley said you might be able to reproduce on other machines.

Yes I can access it. Thanks.

**Comment by e10harvey** [comment]
I think the clean up for #508 at e10harvey/issue-508 just fixed this.

**Comment by seheracer** [comment]
@e10harvey Great! I'd like to try it.

**Comment by e10harvey** [comment]
> @e10harvey Great! I'd like to try it.

I'll flag you in the PR.

**Comment by e10harvey** [comment]
> > @e10harvey Great! I'd like to try it.
> 
> I'll flag you in the PR.

@seheracer: I spoke too soon. e10harvey/issue-508 does not fix this.

**Comment by e10harvey** [comment]
@seheracer: spgemm_jacobi with float has been fixed via 3d1e9ff.

**Comment by brian-kelley** [comment]
Closing this as it has been fixed for a long time.
